troubleshooting/debugging.md

```diff
@@ -1498,5 +1504,114 @@ If you use Amazon ECS FireLens, then standard out and standard error will be col
+
+Above is the rate of errors with an "errno" for all 40 tasks. 
+
+![Rate of all errors for 40 tasks sending to S3](images/all_errors_40_tasks.png)
+
+Above is the rate of all errors for all 40 tasks. 
+
+#### CPU and Memory 
+
+The [Case Studies Warning](#🚨-warning-🚨) is very important for this data; resource usage can vary considerably based on many factors. 
+
+We run a number of different stability test cases against our `cloudwatch_logs` output which cover configurations with non-trivial complexity that are similar to real world cases:
+- [fluent-bit.conf](https://github.com/aws/firelens-datajet/blob/6eaeb5f8e15739e1caf96e2c61236912502da2a3/apps/firelens-stability/templates/golden-path-mountebank-fargate-v01-11-2023/fluent-bit.conf)
+- [fluent-bit-onepod.conf](https://github.com/aws/firelens-datajet/blob/6eaeb5f8e15739e1caf96e2c61236912502da2a3/apps/firelens-stability/templates/golden-path-mountebank-fargate-v01-11-2023/fluent-bit-onepod.conf)
+
+There are test cases at different throughputs, with some running at [10MB/s sustained throughput](https://github.com/aws/firelens-datajet/blob/6eaeb5f8e15739e1caf96e2c61236912502da2a3/apps/firelens-stability/collections/ecs-firelens-stability-tests/golden-path/cases/onepod-high-throughput.json), which is much higher than most real world use cases.
+
+Our CloudWatch Logs tests use a local network mock of the CloudWatch API; thus, these tests are not fully real world as there will be less network latency sending to the mock. 
+
+We also run a number of tests against the S3 output which do send to Amazon S3:
+- [fluent-bit.conf](https://github.com/aws/firelens-datajet/blob/6eaeb5f8e15739e1caf96e2c61236912502da2a3/apps/firelens-stability/templates/s3-fargate-v04-05-2023/fluent-bit.conf)
+
+One of the S3 test cases runs at [30MB/s sustained throughput](https://github.com/aws/firelens-datajet/blob/6eaeb5f8e15739e1caf96e2c61236912502da2a3/apps/firelens-stability/collections/ecs-firelens-stability-tests/s3-stability/cases/s3-throughput-30mbps.json). 
+
+All of tests run on Fargate and are provisioned with [1 vCPU and 4 GB of memory](https://github.com/aws/firelens-datajet/blob/6eaeb5f8e15739e1caf96e2c61236912502da2a3/apps/firelens-stability/templates/golden-path-mountebank-fargate-v01-11-2023/task-definition.json#L159). 
+
+Finally, our tests mostly configure inputs with the default `storage.type memory` and with a [Mem_Buf_Limit 64M](https://github.com/aws-samples/amazon-ecs-firelens-examples/tree/mainline/examples/fluent-bit/oomkill-prevention).
+
+![graph of memory usage](images/stability_test_memory.png)
+
+We see memory usage vary considerably across test cases based on throughput, complexity of the configuration, and other unknown factors. We also typically see that memory can increase slightly over the lifetime of the Fluent Bit container, however, this does not by itself indicate a [memory leak](#memory-leaks-or-high-memory-usage).
```

Comment:
This observation is good. I like this, as this tells the users what they need to pay attention.